google-cloud-core/src/test/java/com/google/cloud/PolicyTest.java

```diff
@@ -104,6 +104,40 @@ public void testBuilder() {
     assertEquals(0, policy.getVersion());
   }
 
+  @Test
+  public void testPolicyOrderShouldNotMatter() {
+    Role role1 = Role.of("role1");
+    Identity identity1 = Identity.user("user1@example.com");
+    Role role2 = Role.of("role2");
+    Identity identity2 = Identity.user("user2@example.com");
+    Policy policy1 =
+        Policy.newBuilder().addIdentity(role1, identity1).addIdentity(role2, identity2).build();
+    Policy policy2 =
+        Policy.newBuilder().addIdentity(role2, identity2).addIdentity(role1, identity1).build();
+    assertEquals(policy1, policy2);
+  }
+
+  @Test
+  public void testPolicyMultipleAddIdentitiesShouldNotMatter() {
+    Role role1 = Role.of("role1");
+    Identity identity1 = Identity.user("user1@example.com");
+    Role role2 = Role.of("role2");
+    Identity identity2 = Identity.user("user2@example.com");
+    Policy policy1 =
+        Policy.newBuilder()
+            .addIdentity(role1, identity1)
+            .addIdentity(role2, identity2)
+            .addIdentity(role2, identity2)
+            .build();
+    Policy policy2 =
+        Policy.newBuilder()
+            .addIdentity(role2, identity2)
+            .addIdentity(role1, identity1)
+            .addIdentity(role1, identity1)
```

Comment:
The dup call `addIdentity(role1, identity1)` is basically a no-op because addIdentity now uses ImmutableSet: https://github.com/googleapis/java-core/pull/174/files#diff-21fde2c27759e0a48638d124df44c36eR263

Reply:
I'm just saying, I think this test would still pass without the `ImmutableSet` change because the equality check is looking for the size of the collection and that each element in the first is contained in the second. If we have 2 different sizes of inputs, we can be sure that we're de-duping.

Reply:
oh, gotcha!

Reply:
For my reference: `If it's really deduping the size of both lists should be equivalent after introducing ImmutableSet<>, otherwise it can pass the size test and contains test`.